### DISCUSSION TITLE
Soft brick / password locked when trying to revert back to stock firmware

### DISCUSSION THREAD

**Discussion by MicaelJarniac** [egzumer/uv-k5-firmware-custom#505] [General]
I saw someone post this on Instagram:

> BE CAREFUL WITH EGZUMER
>
> WHOSMATT IS THE BEST IMO
>
> (UVK5 FIRMWARES)
>
>> Uvk5 troubleshooting update
>>
>> Hey \<name\>. So I figured out how to flash the stock firmware back into the radio and I was met with a lock screen and passcode. I tried all the obvious combos and looked online for a manufacturer pin to input and there's nothing. So they're bricked. I have 2 new ones coming in on Sunday via Amazon and I'll flash the whosmatt on the new ones. I was able to flash it but the passcode stayed but I was able to read the radio via chirp and the settings/ programming seems pretty straightforward so I think most of my issues where probably with the egzumer firmware I was originally running. I'll touch base again when I'm close to up and running. Thanks again

I then asked for more details, as I couldn't find an issue or discussion about it here already, and they said this:

> The problem presented when the user went to program the radio in chirp. The firmware wasn’t interoperable with chirp

I'm not entirely sure what happened with their radio. Still, from what I pieced together, it seems that they tried this modified firmware, and then decided to revert to the original manufacturer's firmware. Somehow this enabled the password lock feature on the original firmware, for which the user did not know the passcode, thus rendering the radio unusable.

Is it possible to factory reset a password-locked radio?

Also, it may be helpful to have some uninstall instructions, if this process can be prone to user error. Something like, "before uninstalling this custom firmware, reset all settings in CHIRP" or similar.

**Comment by AubsUK** [comment]
When you boot the radio to firmware update (hold `PTT` while turning on the radio), I don't believe it loads the firmware which contains the password.

The issue might be related to the original firmware using different memory locations in the EEPROM than the custom firmware(s).  I'd suspect factory resetting the radio after uploading the original firmware would reset the EEPROM memory locations to be compatible with the firmware loaded, including removing the password.  On the Egzumer firmware this can be done by holding `PTT` + `SIDE BUTTON 1` while turning on the radio, but I can't remember if the original firmware had this extra functionality.  Looking at an [Original Manual](https://manuals.plus/quansheng/uv-k5-two-way-radio-manual), it should be just `Menu` >> `Reset` >> `All`.

**Reply by egzumer** [reply]
> The issue might be related to the original firmware using different memory locations

My firmware eeprom structure is backwards compatible with stock firmware, there should be no issues when going back to stock.

**Comment by egzumer** [comment]
I've never heard anyone having this issue, nor I ever experienced that. I did flash the stock firmware back multiple times.
The password is just an 8byte value in the eeprom and can be easily modified, also the CPS software should be able to change it, at least if the password doesn't lock serial communication, I don't know, official release of my firmware has password support disabled.

Looking at the code my firmware writes all bytes 0xFF effectively deleting the password, this happens on every settings save call. So you flash the firmware, change any setting on the radio and the password that was present before should be gone.

The original poster doesn't seem like he know what he is doing, especially when he recommends whosmatt (UV MOD I assume) which is and old, outdated tool with very, very limited capabilities.

**Comment by josephburt** [comment]
I've had issues flashing both of my devices but that stems from my setup and cable I believe. I've had one issue once but was able to flash back to stock firmware and re-flash a device... I would do the following.

- Check your cable first and make sure you have a decent good quality cable.
- Verify the machine you are using and the surroundings. The cheap radios (K5 included) can be susceptible to have a lot of interference which can cause issues.
- Check your radio power and antenna to make sure everything is good.

Not to say you haven't checked but we are all humans and one thing I have learned from this new hobby is you need to all ways check as there is so many things that can effect what you are doing. We all forget that we plugged that one thing in or made a change and forgot about it. Please keep us updated!
